dbms/src/IO/WriteBuffer.h

```diff
@@ -35,8 +35,6 @@ class WriteBuffer : public BufferBase
       */
     inline void next()
     {
-        if (!offset())
-            return;
```

Comment:
It seems `WriteBuffer::nextIfAtEnd()` should only check that the cursor reaches the end of the buffer before calling `WriteBuffer::nextImpl()`.

Reply:
Will it lead to writing zero bytes into file; compressing block of data with zero size, etc?

Reply:
I think you're right and it's better not to leave a `WriteBuffer` with empty size after initialization.
So I've removed my changes in WriteBuffer.h and just corrected formula in `WriteBufferFromVector(AppendModeTag)`.